File: solutions/_2405_optimal_partition_of_string/solution.py
class Solution:
    def partitionString(self, s: str) -> int:
        res = 1
        flag = 0

        for i in range(len(s)):
            val = ord(s[i]) - ord("a")
            if (flag & (1 << val)) != 0:
                flag = 0
                res += 1

            flag = flag | (1 << val)

        return res


# https://leetcode.com/problems/optimal-partition-of-string/

File: solutions/_2405_optimal_partition_of_string/solution.java
package _2405_optimal_partition_of_string;

class Solution {

  public int partitionString(String s) {
    int res = 1;
    int flag = 0;

    for (int i = 0; i < s.length(); i++) {
      int val = s.charAt(i) - 'a';
      if ((flag & (1 << val)) != 0) {
        flag = 0;
        res++;
      }
      flag = flag | (1 << val);
    }

    return res;
  }
}
/*
https://leetcode.com/problems/optimal-partition-of-string/
*/
